Handle different possible plugin dirnames

joncrangle opened the pull request:
Depending on how the plugin is imported:
`local resurrect = wezterm.plugin.require 'https://github.com/MLFlexer/resurrect.wezterm'` or `local resurrect = wezterm.plugin.require 'https://github.com/MLFlexer/resurrect.wezterm/'`, a different directory in the Wezterm plugins folder will be created. If you require the plugin with a `/` at the end, the plugin directory will have `sZs` at the end. You can end up duplicate plugin directories with different names

This PR:
1) checks for the existence of these directories and will return `httpssCssZssZsgithubsDscomsZsMLFlexersZsresurrectsDswezterm` by default 
2) creates an `is_windows()` utility function to use throughout.

MLFlexer commented:
Thanks for the PR!

1.
There is a problem with how sub modules require other sub modules, which I think we need to address. Submodules have the following code in them to be able to depend on submodules. 
```lua
-- in table_state.lua
-- Line below can create a clone of the already existing repo, but with the sZs postfix
local resurrect = wezterm.plugin.require("https://github.com/MLFlexer/resurrect.wezterm/") 
local pane_tree_mod = require(resurrect.get_require_path() .. ".plugin.resurrect.pane_tree")
```
From my perspective, there are 2 options, one is to change the `package.path` to include the path `<plugin_directory>/<name_of_resurrect_directory>/plugin/?.lua`. Then the code for requireing the modules would be changed to this:
```lua
-- in table_state.lua
local pane_tree_mod = require("resurrect.pane_tree")
```
The other option is just keeping what we have, and then users might have a duplicate of the plugins, however maintaining becomes harder as there might be two different versions of the plugin installed.

I hope this problem can be fixed with the next release of Wezterm, but for now this is the solutions as I see them

2.
I really like the `is_windows()` function, it makes the code much more readable

joncrangle commented:
I've updated the PR to add the plugin to the package.path and handle the submodules differently.

>[!WARNING] 
>This represents a breaking change to the API and I've updated the README accordingly.

MLFlexer reviewed (approved):
This is just what I had in mind! Thank you very much for implementing this! Also thanks for updating the docs, i think this is much better and simpler than before 😄
